src/client/game_formspec.cpp

```diff
@@ -270,20 +272,25 @@ void GameFormSpec::showPauseMenuFormSpec(const std::string &formspec, const std:
 	// the in-game settings formspec.
 	// Neither CSM nor the server must be allowed to mess with it.
 
-	if (handleEmptyFormspec(formspec, formname))
+	// If we send updated formspec contents, we can either (1) recycle the old
+	// GUIFormSpecMenu or (2) close the old and open a new one. This is option 2.
+	(void)handleEmptyFormspec("", formname);
+	if (formspec.empty())
 		return;
 
 	FormspecFormSource *fs_src = new FormspecFormSource(formspec);
 	LocalScriptingFormspecHandler *txt_dst =
 		new LocalScriptingFormspecHandler(formname, m_pause_script.get());
 
-	m_formname = formname;
-	GUIFormSpecMenu::create(m_formspec, m_client, m_rendering_engine->get_gui_env(),
```

Comment:
Note to reviewers: `m_formspec` should not exist in the first place. `g_menumgr` manages the layers and we shouldn't track this separately. The PR already ended up a little messy, thus I decided against such cleanup.

Reply:
leave a FIXME comment in the code?

Reply:
[d117cca](https://github.com/luanti-org/luanti/pull/15891/commits/d117cca9cfef5450e301f87a27a30afc0452c9d0)